nginx.montagu.conf

```diff
@@ -49,6 +49,23 @@ server {
     location /reports/ {
         proxy_pass http://report/;
     }
+
+    location /shiny/ {
+        rewrite ^/shiny/(.*)$ /$1 break;
+        
+        proxy_pass http://shiny_proxy;
+        proxy_redirect http://shiny_proxy/ $scheme://$host/shiny/;
+        proxy_http_version 1.1;
+        proxy_set_header Upgrade $http_upgrade;
+        proxy_set_header Connection $connection_upgrade;
+        proxy_read_timeout 20d;
+        proxy_buffering off;
+
+        proxy_set_header    Host $host;
+        proxy_set_header    X-Real-IP $remote_addr;
+        proxy_set_header    X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header    X-Forwarded-Proto $scheme;
```

Comment:
A brief comment explaining why we need to forward these three headers would be helpful too :)